libctru/include/3ds/services/fspxi.h

```diff
@@ -0,0 +1,645 @@
+
+/**
+ * @brief Gets the legacy banner data of a program.
+ * @param mediaType Media type of the program.
+ * @param programId ID of the program.
+ * @param banner Pointer to output the legacy banner data to. (size = 0x23C0)
+ * @param unk Unknown. Always 1?
+ */
+DEFINE_PXIFS_SERVICE_METHOD(GetLegacyBannerData, FS_MediaType mediaType, u64 programId, u8* banner, u32 unk);
+
+/**
+ * Unknown command 3D
+ * @param unk Unknown. Transaction?
+ */
+DEFINE_PXIFS_SERVICE_METHOD(0x3D, u32 unk);
+
+/// Deletes the 3DS SDMC root.
+DEFINE_PXIFS_SERVICE_METHOD(DeleteSdmcRoot, void);
+
+/// Deletes all ext save data on the NAND.
+DEFINE_PXIFS_SERVICE_METHOD(DeleteAllExtSaveDataOnNand, void);
+
+/// Initializes the CTR file system.
+DEFINE_PXIFS_SERVICE_METHOD(InitializeCtrFilesystem, void);
+
+/// Creates the FS seed.
+DEFINE_PXIFS_SERVICE_METHOD(CreateSeed, void);
+
+/**
+ * @brief Gets the CTR SDMC root path.
+ * @param out Pointer to output the root path to.
+ * @param length Length of the output buffer.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(GetSdmcCtrRootPath, u8* out, u32 length);
+
+/**
+ * @brief Gets an archive's resource information.
+ * @param archiveResource Pointer to output the archive resource information to.
+ * @param mediaType System media type to check.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(GetArchiveResource, FS_ArchiveResource* archiveResource, FS_SystemMediaType mediaType);
+
+/**
+ * @brief Exports the integrity verification seed.
+ * @param seed Pointer to output the seed to.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(ExportIntegrityVerificationSeed, FS_IntegrityVerificationSeed* seed);
+
+/**
+ * @brief Imports an integrity verification seed.
+ * @param seed Seed to import.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(ImportIntegrityVerificationSeed, FS_IntegrityVerificationSeed* seed);
+
+/**
+ * @brief Gets the legacy sub banner data of a program.
+ * @param bannerSize Size of the banner.
+ * @param mediaType Media type of the program.
+ * @param programId ID of the program.
+ * @param header Pointer to output the legacy sub banner data to.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(GetLegacySubBannerData, u32 bannerSize, FS_MediaType mediaType, u64 programId, u8* banner);
+
+/// Unknown command 47
+DEFINE_PXIFS_SERVICE_METHOD(0x47, void* buf, u32 size);
+
+/**
+ * @brief Gets the last modified time of a file in an archive.
+ * @param archive The archive that contains the file.
+ * @param out The pointer to write the timestamp to.
+ * @param path The UTF-16 path of the file.
+ * @param size The size of the path.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(GetFileLastModified, FSPXI_Archive archive, u64* out, u16* path, u32 size);
+
+/**
+ * @brief Reads from a special file.
+ * @param bytesRead Pointer to output the number of bytes read to.
+ * @param fileOffset Offset of the file.
+ * @param size Size of the buffer.
+ * @param data Buffer to read to.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(ReadSpecialFile, u32* bytesRead, u64 fileOffset, u32 size, u8* data);
```

Comment:
```suggestion
DEFINE_PXIFS_SERVICE_METHOD(ReadSpecialFile, u32* bytesRead, u64 fileOffset, u32 size, void* data);
```

Reply:
I went with a u8* here because FS's ReadSpecialFile has a u8*. Should I change that there, too?

Reply:
Yeah, it's probably wrong over there too.